src/scale/hierarchical.js

```diff
@@ -159,22 +163,29 @@ const HierarchicalScale = Scale.extend({
       }
     }
 
-    const node = this._nodes[index];
-    const centerTick = this.options.offset;
-    const base = this.isHorizontal() ? this.left : this.top;
-
-    return base + node.center - (centerTick ? 0 : node.width / 2);
+    return this._centerBase(index);
   },
 
   getPixelForTick(index) {
     if (index === 1 && this._nodes.length === 1) {
       // cornercase in chartjs to determine tick with, hard coded 1
       return this._nodes[0].width;
     }
+
+    return this._centerBase(index);
+  },
+
+  _centerBase(index) {
     const centerTick = this.options.offset;
-    const node = this._nodes[index + this.minIndex];
     const base = this.isHorizontal() ? this.left : this.top;
-    return base + node.center - (centerTick ? 0 : node.width / 2);
+    const node = this._nodes[index + this.minIndex];
```

Comment:
This code is equal to the one from `getPixelForTick()`, but differs from the one in `getPixelForValue()`. Is this working correctly for the call in `getPixelForValue()`?

Call from `getPixelForTick()`:

```ts
const node = this._nodes[index + this.minIndex];
```

Call from `getPixelForValue()`:

```ts
const node = this._nodes[index];
```